src/routes/overviews/bufnmf.svx

```diff
@@ -0,0 +1,134 @@
+---
+title: Audio Decomposition using BufNMF
+blurb: |
+    Decompose an audio buffer into component parts using non-negative matrix factorization.
+tags:
+    - decomposition
+flair: article
+---
+
+<script>
+    import Image from '$lib/components/Image.svelte';
+    import Audio from '$lib/components/Audio.svelte';
+</script>
+
+# Decomposition
+
+"NMF" stands for Non-negative Matrix Factorization, which is a machine learning algorithm that can be used for decomposing an audio buffer into component parts. In this example, we’ll look at decomposing a mono drum loop into three different mono audio buffers, each hopefully containing the hits from just one of the instruments: kick, snare, and cymbal. Here is the recording and spectrogram of our drum loop.
+
+<Audio
+src="/audio/Nicol-LoopE-M.mp3"
+label="Mono drum loop included in the FluCoMa package."
+/>
+
+<Image
+src="/overviews/bufnmf/00_original_spectrogram_mel_color.png"
+label="Mel-frequecy spectrogram of drum loop"
+/>
+
+The spectrogram below roughly identifies where in time and frequency each of these instruments are present.
+
+<Image
+src="/overviews/bufnmf/00_boxes_over_nicol-01.png"
+label="(red = kick, blue = snare, yellow = cymbal)"
+/>
+
+As you can see (and hear in the playback), these different drum hits overlap quite a lot in both time and spectrum meaning that a purely time-based (such as a slicer) or purely spectral (such as a filter) strategy for separating these hits wouldn’t work very well.
+
+This is where Non-negative Matrix Factorization can be quite useful.
+
+We can use BufNMF to analyze the drum loop’s spectrogram and determine which parts of the spectrum always (or often) occur together in time. For example, in the snare hits (the green boxes) you can see the different partials of the drum head’s resonant tone (the horizontal bands from ~300 Hz to ~2000 Hz) always occurring together. Sometimes there is also some cymbal sound happening (above that in a red box), but because the cymbal sound is often not there, NMF will try to decompose the spectrogram in such a way that these can be considered separate components. In general, the more that different sound objects are separated out in time and the more repetitions of each sound object, the better BufNMF will be at identifying and separating them.
+
+Processing this mono drum loop with BufNMF (specifying that we want it to try to decompose the sound into three components) returns these three buffers, one with each component it was able to separate:
+
+<a href='/overviews/bufnmf/bufnmf-overview-example.scd' download='bufnmf-overview-example.scd'>SuperCollider Example</a>
+<a href='/overviews/bufnmf/bufnmf-overview-example.maxpat' download='bufnmf-overview-example.maxpat'>Max Example</a>
+
+### Component \#1 (contains mostly just the kick drum hits)
+
+<Image
+src="/overviews/bufnmf/05_resynth_spectrogram_mel_component_0.png"
+label=""
```

Comment:
I think that all images should have labels. They look floaty without them and it helps screen readers (because it provides the alt tag).